components/shared/base/lib.rs

```diff
@@ -14,12 +14,62 @@ pub mod generic_channel;
 pub mod id;
 pub mod print_tree;
 pub mod text;
+pub mod threadpool;
 mod unicode_block;
 
+use std::fs::File;
+use std::io::{BufWriter, Read};
+use std::path::Path;
+
+use ipc_channel::ipc::{IpcError, IpcSender};
+use log::{trace, warn};
 use malloc_size_of_derive::MallocSizeOf;
 use serde::{Deserialize, Serialize};
 use webrender_api::Epoch as WebRenderEpoch;
 
+pub fn read_json_from_file<T>(data: &mut T, config_dir: &Path, filename: &str)
```

Comment:
It's nice to see utility functions can be shared but will there be any other place using it except net crate?

Reply:
Yes, the new storage crate proposed by #39418.